salvo/ENVOY_DEVELOP_WORKFLOW.md

```diff
@@ -0,0 +1,135 @@
+# Measure Envoy's Performance Change with an A/B Testing
+
+## Objective 
+
+This documentation will describe the user journey and detailed workflow to use Salvo to measure Envoy's performance change with an A/B testing.
+
+## User Journey
+
+You are implementing a new feature in Envoy and reach a point where you need to decide between two possible implementation approaches. You wonder what are the performance implications of each choice and decide to run the benchmarking framework to find out. You run two executions in A/B testing mode, both comparing an official Envoy version to an Envoy version with your change using one of the approaches. Once the tests complete, you examine the results and choose an implementation approach with the better performance characteristics.
+
+## Workflow
+
+This section will describe the whole workflow to use Salvo to Measure Performance Change with an A/B Testing. And we will use [Binary Benchmark](./README.md#binary-benchmark) as an example.
+
+1. The requirements to build Envoy and Nighthawk in local machine
+
+    In oder to build Envoy and Nighthawk in local machine, follow [this documentation](https://www.envoyproxy.io/docs/envoy/latest/start/building.html#linux-mac-target-requirements) to ensure local machine meets the requirements. 
+
+2. Download Salvo
+
+    The next step is to download Salvo project from github, you can use `git clone` command:
+
+    ```bash
+    git clone https://github.com/envoyproxy/envoy-perf.git
+    cd envoy-perf/salvo
+    ```
+
+3. Install dependencies 
+
+    Before using Salvo, you need to install its dependencies. The [install_deps.sh](./install_deps.sh) script can be used to install any dependencies required by Salvo. 
+
+    ```bash
+    ./install_deps.sh
+    ```
+
+4. Build Salvo
+
+    The next step is to build Salvo, Salvo can be built by [Bazel](https://docs.bazel.build/versions/main/install-ubuntu.html):
+
+    ```bash
+    bazel build //...
+    ```
+
+5. Binary Benchmark Configuration
+    
+    For an example, you committed two Envoy code changes to your forked repo "https://github.com/gyohuangxin/envoy" and the commit hashes were "c2e1c0d0da61f3fec25f48f8bd63e4a76db8a2a9" and "e95717bcd0672910826de4a9bb2f402f6e057b2c". Then you want to measure their performance change to Envoy v1.21.0 as the baseline and choose the commit which has the better performance. 
```

Comment:
Yes, this example commits are not real, because it's hard to find a commit changing the performance significantly unless we have many commits' benchmarking test results. So I use the two virtual commits as the example. Do you know which two commits will be the best example? Or we can just choose at will?

Reply:
Both of these commits claim to improve the performance of the H1 protocol by ~5 percent. Maybe we can experiment with them, or choose a commit before and after both of them?

We should also make sure that the test this is executing exercises H1.

- https://github.com/envoyproxy/envoy/pull/19115
- https://github.com/envoyproxy/envoy/pull/20148

Should we give this a try? Reproducing the performance gain would also help us to gain some experience with how to use Salvo and might uncover some Salvo bugs.

Reply:
@mum4k Thank you for finding out these two commits, of course, I will give a try. And I agreed with you it will help us understand more about Salvo and where we can improve.

Reply:
As discussed, we will do this in a follow-up PR.